vncviewer/UserDialog.cxx

```diff
@@ -62,34 +63,54 @@ static void button_cb(Fl_Widget *widget, void *val) {
   widget->window()->hide();
 }
 
+static void keepPassword(Fl_Widget *widget, void *data) {
+  UserDialog *dialog = (UserDialog*) data;
+  Fl_Check_Button* flcb = (Fl_Check_Button*)widget;
+  dialog->setKeepPassword(flcb->value());
+}
+
 UserDialog::UserDialog()
 {
+  user_ = NULL;
+  password_ = NULL;
+  keepPasswd_ = true;
```

Comment:
Just a FYI, in C++ it's usually better to use initialisers for things like this. I.e.:

```c++
UserDialog::UserDialog() : user_(NULL), password_(NULL), keepPasswd_(true)
{
}
```